java/piranha/src/main/java/com/uber/piranha/config/Config.java

```diff
@@ -124,16 +126,33 @@ private Config(
   }
 
   /**
-   * Return all configuration method records matching a given method name.
+   * Return all configuration method records matching a given method name. A method record name
+   * field can be used to express a single method invocation or a chain of two method invocations. A
+   * chain of method invocations can be expressed by joining the name of the chained methods (from
+   * left to right) with a `.`. For instance, the name `isToggle` will match a invocation like -
+   * :[1].isToggle(:[2])), and the name `stale_flag.getValue` will match a invocation like -
+   * :[1].stale_flag(:[2]).getValue(:[3]).
    *
-   * @param methodName the method name to search
+   * @param mit Method invocation AST
    * @return A collection of {@link PiranhaMethodRecord} objects, representing each method
    *     definition in the piranha json configuration file matching {@code methodName}.
    */
-  public ImmutableCollection<PiranhaMethodRecord> getMethodRecordsForName(String methodName) {
-    return configMethodProperties.containsKey(methodName)
-        ? configMethodProperties.get(methodName)
-        : ImmutableSet.of();
+  public ImmutableCollection<PiranhaMethodRecord> getMethodRecordsForName(
+      MethodInvocationTree mit) {
+    MemberSelectTree mst = (MemberSelectTree) mit.getMethodSelect();
+    String methodName = mst.getIdentifier().toString();
+    if (configMethodProperties.containsKey(methodName))
+      return configMethodProperties.get(methodName);
+    if (mst.getExpression() instanceof MethodInvocationTree) {
```

Comment:
since this scenario will be infrequent for most users of Piranha, can we add a quick filter (say an option that is false by default) to handle "chaining of methods"?